lib/open_project/translations/models/combined_locales_updater.rb

```diff
@@ -0,0 +1,313 @@
+      download_locales do |zip_file|
+        all_locale_paths.each do |module_dir|
+          mod_name = get_crowdin_name(module_dir)
+          path = versioned_path(mod_name)
+          entries =  zip_file.glob("*/#{path}/*.yml")
+          Dir.chdir(module_dir) { replace_locales(module_dir, entries) }
+        end
+      end
+    end
+  end
+
+  private
+
+  ##
+  # Download the latest translations build from crowdin
+  def download_locales
+    Tempfile.create 'crowdin_translations' do |file|
+      crowdin.download_translation 'all', output: file.path
+      Zip::File.open(file.path) do |zip_file|
+        yield zip_file
+      end
+    end
+  end
+
+  ##
+  # Override local crowdin locales
+  def replace_locales(mod_path, entries)
+    mod_name = get_crowdin_name(mod_path)
+
+    target_directory = Pathname(File.join('crowdin'))
+    unless File.directory?(target_directory)
+      FileUtils.mkdir_p(target_directory)
+    end
+
+    # Clear all locales before checking in the current ones
+    FileUtils.rm_f Dir.glob("#{target_directory}/*.yml")
+
+    entries.each do |entry|
+      language_name = entry.name.split('/').first # the file is put in a directory containing the language name
+
+      # only take translations with enough percent translated
+      next unless translation_status_high_enough?(language_name, ACCEPTANCE_LEVEL)
+
+      filepath = target_directory.join "#{js_translation?(Pathname.new(entry.name)) ? 'js-' : ''}#{language_name}.yml"
+      replace_file(filepath, entry)
+    end
+  end
+
+  ##
+  # Replace a single translation file from crowdin
+  def replace_file(filepath, new_file)
+    File.delete(filepath) if File.file?(filepath)
+
+    contents = new_file.get_input_stream.read
+
+    # work around a crowdin bug which does not escape norwegian key
+    # and results in boolean
+    language_key = calculate_language_key filepath
+    if language_key == 'no'
+      contents.gsub! /\Ano:/, '"no":'
+    end
+
+    File.open(filepath, 'wb') do |file|
+      file.write contents
+    end
+  end
+
+  ##
+  # Determine and fix the localization language key from the file path
+  def calculate_language_key(translation_file_path)
+    # The main translation key is the language key ('en' for english, 'zh' for chinese etc.)
+    # Sometimes a language has multiple variants (eg simplified chinese 'zh-TW')
+    # For language variants ('zh-TW') crowdin gives us only the language key ('zh') so we replace it.
+    # from /path/to/openproject-translations/config/locales/zh-TW.yml we extract 'zh-TW'
+    language_key = if js_translation?(translation_file_path)
+      # ignore the 'js-' prefix of js translations
+      translation_file_path.basename.to_s[3..-5]
+    else
+      translation_file_path.basename.to_s[0..-5]
+    end
+
+    # Unfortunately OpenProject has some vendored translations in jstoolbar, which
+    # needs different names sometimes, so we map exceptions here
+    mapping = Hash.new {|hash, key| key }
```

Comment:
Given the comment, I assume this is no longer needed. Don't know if it hurts, though and I am aware that it has just moved.

Reply:
I have not tried whether we can remove it and would vote for leaving it in